tests/perf/bench/test_runtest_hook.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import absolute_import, division, print_function
+
+import logging
+
+import pytest
+
+
+logger = logging.getLogger('pytest_catchlog.test.perf')
+
+
+@pytest.fixture(autouse=True)
+def bench_runtest(request, benchmark):
+    # Using benchmark.weave to patch a runtest hook doesn't seem to work with
+    # pytest 2.8.3; for some reason hook gets called more than once, before
+    # running benchmark cleanup finalizer, resulting in the
+    # "FixtureAlreadyUsed: Fixture can only be used once" error.
+    #
+    # Use plain old monkey patching instead.
+    ihook = request.node.ihook
+    saved_hook = ihook.pytest_runtest_call
+
+    def patched_hook(*args, **kwargs):
+        ihook.pytest_runtest_call = saved_hook  # restore early
+        return benchmark(saved_hook, *args, **kwargs)
+
+    ihook.pytest_runtest_call = patched_hook
```

Comment:
Yes, this will benchmark the tests defined in this file only (am I right? this won't apply to tests from other files, will it?).

Here we want to measure the cost of preparing our `caplog` fixture and then compare that with an overhead of having an empty stub fixture. Hence, two test cases, both being benchmarked.